### DISCUSSION TITLE
Cask: beginners question about updating alt-tab cask

### DISCUSSION THREAD

**Discussion by creativeusernamegoeshere** [Homebrew/discussions#5665] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: 84c31175f11860129a9aaed40a13c549625e2db1
Last commit: 9 days ago
Core tap JSON: 10 Oct 09:28 UTC
Core cask tap JSON: 10 Oct 09:28 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: dodeca-core 64-bit arm_lobos
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.0.1-arm64
CLT: 16.0.0.0.1.1724870825
Xcode: 16.0
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

https://formulae.brew.sh/cask/alt-tab#default shows `Current version: 7.0.0`.

When checking via CLI:

```
➜  ~ brew list --cask --versions alt-tab
alt-tab 6.72.0
```

This persists after `brew update`. However when  running `brew upgrade --cask alt-tab` seeing `Upgrading 1 outdated package:
alt-tab 6.72.0 -> 7.0.0`.

Should this not have been seen after  `brew update` already?

### Relevant casks

alt-tab

**Comment by SMillerDev** [comment]
`brew update` only updates brew, `brew upgrade` upgrades packages you have installed.

**Comment by creativeusernamegoeshere** [comment]
Thanks for the quick reply. Right, but that did not update alt-tab. My question is, should `brew upgrade` have updated alt-tab to v7.0.0 when executed? (It did not)

```
➜  ~ brew update                
==> Updating Homebrew...
Already up-to-date.
```

and

```
brew upgrade               
==> Upgrading 2 outdated packages:
python-packaging 24.1 -> 24.1_1
openexr 3.3.0 -> 3.3.1
```

but alt-tab 7.0.0 was not found.

**Comment by carlocab** [comment]
`alt-tab` auto-updates, so `brew` doesn't upgrade it to a new version with `brew upgrade`. If you want `brew` to upgrade it anyway:
```
brew upgrade --greedy alt-tab
```
You can see the fact that `alt-tab` auto-updates with `brew info alt-tab`:
```
❯ brew info alt-tab
==> alt-tab: 7.0.0 (auto_updates)
[snip]
```
`brew` most likely also mentioned this when you did `brew install alt-tab`.

**Comment by creativeusernamegoeshere** [comment]
Not familiar with the internals but what is the reasoning behind using auto-updates, if they don't ... well, auto-update and require a manual trigger for the specific app in order to find an update? Or am I misunderstanding something?

**Reply by carlocab** [reply]
Casks that auto-update do not require a manual trigger to find an update. `brew upgrade --greedy` is just there if you want to ignore the cask's auto-update mechanism and update it using `brew` anyway.
